### PR TITLE
Ab#90324 Improve filtering on datetime fields

### PR DESCRIPTION
**Pull request by MwanPygmay**
# Description

Fix {{today}} semantic broken by https://github.com/ReliefApplications/ems-backend/pull/919/files.

## Useful links

- [Please insert link to ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/90324)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Created a calculated field that converts createdAt to a date (today will not work with datetime). created a filter based on {{today}} and the calculated field.

Retried Yafar's tests from above PR: when testing equality, I do not get any record, but this seems coherent with the code ? Not sure though: 
image
See below gif: 
retestYafar

# Checklist:

( * == Mandatory ) 

- [x] * I have set myself as assignee of the pull request
- [x] * My code follows the style guidelines of this project
- [x] * Linting does not generate new warnings
- [x] * I have performed a self-review of my own code
- [x] * I have put the ticket for review, adding the oort-backend team to the list of reviewers
- [x] * I have commented my code, particularly in hard-to-understand areas
- [x] * I have put JSDoc comment in all required places
- [x] * My changes generate no new warnings
- [x] * I have included screenshots describing my changes if relevant
- [x] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# More explanation
https://www.loom.com/share/05a716d61b9744faaf51fb304c21d1e5?sid=f87cf896-582a-4f76-93ae-8ceed801b145

### CODE REVIEW

**Comment by AntoineRelief** [comment]
@MwanPygmay 

Indeed the "eq" works with day + time so it's hard to exactly get what you want
but I think I'll edit the logic a bit to allow the today to work

**Comment by AntoineRelief** [comment]
@MwanPygmay 

I think we should add a custom method for datetime, so:
- {{now}} works as it is right now
- {{today}} also works, as it works for dates ( removing the datetime so )
- dates like that: 2024-04-17 can work ( this is what we pass through the UI when setting default filters in layouts )

File: src/utils/schema/resolvers/Query/getFilter.ts [hunk 1]
```diff
@@ -222,12 +227,24 @@ const buildMongoFilter = (
...
-            // startDatetime contains the beginning of the minute
-            startDatetime = getTimeForMongo(value);
-            // endDatetime contains the end of the minute (last second, last ms)
-            endDatetime = new Date(startDatetime.getTime() + 59999);
-            // we end up with a date range covering exactly the minute selected,
-            // regardless of the saved seconds and ms
+            //if we are using the {{today}} operator
+            if (
+              value === Placeholder.TODAY ||
+              REGEX_TODAY_MINUS.test(value) ||
```
Comment (AntoineRelief) [inline src/utils/schema/resolvers/Query/getFilter.ts:233]: @MwanPygmay 
perhaps you should put that in a custom method, as such logic would also be needed in the other file called "getFilter"

**Comment by MwanPygmay** [comment]
@Matthis-M-ReliefApps

**Pull request merged**